### ISSUE TITLE
How to specify temp files

### ISSUE THREAD

**Issue by ziqihugo** [flatironinstitute/CaImAn#1419]
Please fill in the following for any issues

Your setup:
 1. Operating System (Linux, MacOS, Windows): Windows
 2. Hardware type (x86, ARM..) and RAM: x86
 3. Python Version (e.g. 3.9): 3.12.3
 4. Caiman version (e.g. 1.9.12): 1.11.3
 5. Which demo exhibits the problem (if applicable):
 6. How you installed Caiman (pure conda, conda + compile, colab, ..): conda
 7. Details: I am running a batch process for motion correction. The problem is that every time I run the batch process, caiman generates a series of temp files under 'temp' of 'Caiman_data' in C drive. I have limited space on my laptop. Is there a way to specify the location of temp files so that I can store those huge files in my external hard drive during the batch process?

**Comment by programerkotik** [comment]
I’m encountering a similar issue with motion correction and CNMF-E while running batch analyses. Despite having extensive resources (including terabytes of hard drive space), the process still manages to fill up all available storage. I don’t understand why CaImAn isn’t cleaning up temporary files once processing for each file is complete. Is there a way to enforce this cleanup behaviour?

**Comment by pgunn** [comment]
Hello, probably the best route is to set CAIMAN_DATA to a path on that external storage (and rerun `caimanmanager install` after doing so); if you always set the env var that way, it should keep all data on your larger storage.

@programerkotik The temporary files are not cleaned up for two reasons - first, there are potential uses for them, and second, right now caiman doesn't have an enforced workflow where it behaves like an application; instead people can call into caiman via the APIs, it will do work, and they can keep calling those APIs for as long as they like, with some of them needing those temporary files. I have some ideas in mind to potentially change some of this, and the notebooks and CLI demos probably could also give examples of what a good cleanup looks like, but at least the present design assumes people have plenty of storage.

**Comment by programerkotik** [comment]
Thanks a lot! Yes, I just introduced the file transfer to my external storage. It seems that it works.

**Comment by ziqihugo** [comment]
> Hello, probably the best route is to set CAIMAN_DATA to a path on that external storage (and rerun `caimanmanager install` after doing so); if you always set the env var that way, it should keep all data on your larger storage.
> 
> @programerkotik The temporary files are not cleaned up for two reasons - first, there are potential uses for them, and second, right now caiman doesn't have an enforced workflow where it behaves like an application; instead people can call into caiman via the APIs, it will do work, and they can keep calling those APIs for as long as they like, with some of them needing those temporary files. I have some ideas in mind to potentially change some of this, and the notebooks and CLI demos probably could also give examples of what a good cleanup looks like, but at least the present design assumes people have plenty of storage.

I actually found another way:

os.environ['CAIMAN_TEMP'] = 'where you want to store the temp files'

**Comment by steinkki** [comment]
> Thanks a lot! Yes, I just introduced the file transfer to my external storage. It seems that it works.

Hello! I am new to caiman (and an amateur at python honestly) and am trying desperately to figure out how to set the Caiman_data path to external storage... please how did you accomplish this?? "set the environment variable" is unfortunately to vague for me to understand. Thank you!!

**Comment by pgunn** [comment]
@steinkki What OS are you on?

**Comment by steinkki** [comment]
@pgunn I am on Linux. 

Through looking at other discussions, I was able to cobble together two possible solutions: 

1) add the line os.environ['CAIMAN_DATA']='/path/to/new/data/directory' to the demo_pipeline.ipynb 
-or-
2) add export CAIMAN_DATA=/path/to/new/data/directory to the activate.d file in the caiman env.

they both appear to work when I tried them, but is one solution better/recommended over the other? 

Sorry for such novice questions, and thank you for your time!

PS I will copy this convo (or link if I can figure that out) to another discussion board on here where some others had the same question as me, so you hopefully don't have to reply the same thing twice!!
